packages/core/src/odata/v4/entity-deserializer.ts

```diff
@@ -1,8 +1,42 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
 
-import { enityDeserializer } from '../common/entity-deserializer';
+import { entityDeserializer } from '../common/entity-deserializer';
+import {
+  ComplexTypeField,
+  EdmTypeField,
+  EntityBase,
+  Field,
+  Link
+} from '../common';
 import { edmToTs } from './payload-value-converter';
-const deserializer = enityDeserializer(edmToTs);
+import { CollectionField } from './selectable/collection-field';
+const deserializer = entityDeserializer(edmToTs, getFieldValueV4);
 
 export const extractCustomFields = deserializer.extractCustomFields;
 export const deserializeEntity = deserializer.deserializeEntity;
+
+function getFieldValueV4<EntityT extends EntityBase, JsonT>(
+  json: Partial<JsonT>,
+  field: Field<EntityT> | Link<EntityT>
+) {
+  if (deserializer.isODataV2Field(field)) {
+    return deserializer.getFieldValue(json, field);
+  } else if (field instanceof CollectionField) {
+    return deserializeCollectionType(json[field._fieldName], field);
+  }
+}
```

Comment:
This function confuses me a lot:

1. First of all, I would expect `deserializer.isODataV2Field(field)` to always return false in the context of a function called `getFieldValueV4` - The field should never be a v2 field. I understand that this could probably be improved by different naming.
2. This is meant to add deserialization for collection types. I think we already discussed this and decided to add it to the original `getFieldValue` function https://github.com/SAP/cloud-sdk/blob/319be49d7dfe80f92811423375417fd7844bdab2/packages/core/src/odata/common/entity-deserializer.ts#L102. This would make the code simpler (no need to pass functions back to the common converter) and it does not interfere with the v2 implementation. In addition this can be used for wrong v2 implementations, that allow to have collections (e. g. SFSF).